packages/zoe/src/contractSupport/prepare-ownable.js

```diff
@@ -0,0 +1,89 @@
+import { M } from '@endo/patterns';
+import { OfferHandlerI } from '../typeGuards.js';
+import { prepareRevocableKit } from './prepare-revocable.js';
+
+const TransferProposalShape = M.splitRecord({
+  give: {},
+  want: {},
+  exit: {
+    onDemand: null,
+  },
+});
+
+/**
+ * @template {any} [U=any]
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {MakeInvitation} makeInvitation
+ * @param {string} uKindName
+ * @param {string} uInterfaceName
+ * @param {(string|symbol)[]} uMethodNames
+ * @returns {(underlying: U) => U}
+ */
+export const prepareOwnable = (
+  zone,
+  makeInvitation,
+  uKindName,
+  uInterfaceName,
+  uMethodNames,
+) => {
+  const makeRevocableKit = prepareRevocableKit(
+    zone,
+    uKindName,
+    uInterfaceName,
+    uMethodNames,
+    {
+      makeTransferInvitation: M.call().returns(M.promise()),
+    },
+    {
+      makeTransferInvitation() {
+        const {
+          state: {
+            // @ts-expect-error `this` typing
+            underlying,
+          },
+          facets: {
+            // @ts-expect-error `this` typing
+            revoker,
+          },
+        } = this;
```

Comment:
prettier goes a little nutso with vertical space. Consider:

```suggestion
        const { underlying } = this.state;
        const { revoker } = this.facets;
```

Reply:
Done